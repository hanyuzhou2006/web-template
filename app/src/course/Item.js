import React, { Component } from 'react';
export default class Item extends Component {

    render() {
        const { avatar = "", name = "", actived = false, onClick } = this.props;

        const avatarStyle = avatar ? { backgroundImage: `url(${avatar})` } : {};

        return (
            <div className={`entity ${actived ? "active" : ""}`} onClick={onClick}>
                <span className="p1">
                    <div className="person-icon" style={avatarStyle}></div>
                </span>
                <span className="p2">
                    <span className="name">{name}</span>
                </span>
            </div>
        )
    }
}