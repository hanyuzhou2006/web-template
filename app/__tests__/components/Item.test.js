import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import Item from '../../src/course/Item';

test('render normal', () => {
    const component = renderer.create(
        <Item avatar="/images/1.jpg" name="LiuXia@sansi.com" />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
})

test('render actived', () => {
    const component = renderer.create(
        <Item avatar="/images/1.jpg" name="LiuXia@sansi.com" actived={true} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
})

test('vdom render normal', () => {
    const onClick = jest.fn();
    const item = mount(
        <Item avatar="/images/1.jpg" name="LiuXia@sansi.com" onClick={onClick} />
    );
    const avatar = item.find('.person-icon');
    expect(avatar.prop('style').backgroundImage).toBe('url(/images/1.jpg)');

    const name = item.find('.name');
    expect(name.text()).toBe('LiuXia@sansi.com');

    const actived = item.find('.entity.active');
    expect(actived.length).toBe(0);

    const entity = item.find('.entity');
    entity.simulate('click');
    expect(onClick).toBeCalled();
})

test('vdom render actived', () => {
    const item = mount(
        <Item avatar="/images/1.jpg" name="LiuXia@sansi.com" actived={true} />
    );
    const actived = item.find('.entity.active');
    expect(actived.length).toBe(1);
})