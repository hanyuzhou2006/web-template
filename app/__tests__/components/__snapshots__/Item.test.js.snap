exports[`test render actived 1`] = `
<div
  className="entity active"
  onClick={undefined}>
  <span
    className="p1">
    <div
      className="person-icon"
      style={
        Object {
          "backgroundImage": "url(/images/1.jpg)",
        }
      } />
  </span>
  <span
    className="p2">
    <span
      className="name">
      LiuXia@sansi.com
    </span>
  </span>
</div>
`;

exports[`test render normal 1`] = `
<div
  className="entity "
  onClick={undefined}>
  <span
    className="p1">
    <div
      className="person-icon"
      style={
        Object {
          "backgroundImage": "url(/images/1.jpg)",
        }
      } />
  </span>
  <span
    className="p2">
    <span
      className="name">
      LiuXia@sansi.com
    </span>
  </span>
</div>
`;
